ml-agents/mlagents/trainers/models.py

```diff
@@ -491,54 +491,40 @@ def create_observation_streams(
             self.visual_in.append(visual_input)
         vector_observation_input = self.create_vector_input()
 
+        # Pick the encoder function based on the EncoderType
+        create_encoder_func = LearningModel.create_visual_observation_encoder
```

Comment:
Note that neither the old or new versions raise an error on an invalid enum, they just fall through to `create_visual_observation_encoder`. Could do this as a dict from EncoderType -> Callable but didn't seem like much advantage for this small number of choices.